From eaed059ebb89fb3149a9cb64f7ed9df35cae27e2 Mon Sep 17 00:00:00 2001
From: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
Date: Wed, 30 Oct 2024 18:35:52 -0300
Subject: [PATCH] build/config/compiler/BUILD.gn: Disable CREL

This feature is not supported on the current clang toolchain version,
which causes build to fail with the following error:

| clang++: error: unsupported argument '--crel' to option '-Wa,'
| clang++: error: unsupported argument '--allow-experimental-crel'
  to option '-Wa,'

Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
---
 build/config/compiler/BUILD.gn | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 50b032cc03436..b968616f8e80d 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -597,14 +597,6 @@ config("compiler") {
     } else {
       cflags += [ "-ffp-contract=off" ]
     }
-
-    # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF
-    # (excluding toolchains that use an older version of LLVM).
-    # TODO(crbug.com/376278218): This causes segfault on Linux ARM builds.
-    if (is_linux && !llvm_android_mainline && current_cpu != "arm" &&
-        default_toolchain != "//build/toolchain/cros:target") {
-      cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
-    }
   }

   # C11/C++11 compiler flags setup.
--
2.45.2

